### ISSUE TITLE
Test and eval are not right! model(x) in eval mode output all nan.

### ISSUE THREAD

**Issue by beckhamchen** [eriklindernoren/PyTorch-YOLOv3#89]
The training stage seems good. But when detect and test, the output of model(x) are all nan even after the first conv layer? Anybody know why?

**Comment by Aslizy** [comment]
I have the same problem.

**Comment by zwbdmm** [comment]
I have the same issue, the detect.py doesn't work on the newly trained net. output 0s, through the train stage looks good.

**Comment by jclevesque** [comment]
The save_weight and load_weight functions are not saving/loading all weights. Some are left uninitialized, which most likely causes a division by zero somewhere. A workaround would be to save and load the state_dict instead of manually going through layers.

**Comment by rollthedice32** [comment]
thanks for the suggestion @jclevesque this makes sense, I'll try this out today and report if the nan's go away.

**Comment by tyathalae** [comment]
Thanks @jclevesque, it works in PyTorch 1.0. I wonder how it worked with load_weights at the first place. Maybe only on PyTorch 0.4.

**Comment by eriklindernoren** [comment]
Great observation @jclevesque . I have changed to saving and loading state dicts instead in master.This seems to solve the issue with checkpoint models producing junk when evaluated. There is probably an issue in `Darknet.save_weights`.

**Comment by SKRohit** [comment]
@eriklindernoren I am getting `nan` values even with pre-trained yolo weights without any finetuning. Is it because I am trying to detect objects which are not in MS COCO dataset?

**Comment by SKRohit** [comment]
@eriklindernoren I could solve this issue by running my model in `train` mode. I have this issue only in `eval` mode. But could you please suggest why is it happening?

**Comment by LordonCN** [comment]
me too...
